### PR TITLE
Add support for DROP SCHEMA [IF EXISTS] name [ CASCADE | RESTRICT ]

### PR DESCRIPTION
**Pull request by gimmyxd**
This commit updates the ANSI dialect and adds support for
` DROP SCHEMA [IF EXISTS] name [ CASCADE | RESTRICT ]`

fixes  https://github.com/sqlfluff/sqlfluff/issues/1859

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -2250,8 +2250,9 @@ class DropSchemaStatementSegment(BaseSegment):
     match_grammar = Sequence(
         "DROP",
         "SCHEMA",
-        Ref("IfNotExistsGrammar", optional=True),
+        Ref("IfExistsGrammar", optional=True),
         Ref("SchemaReferenceSegment"),
+        OneOf("RESTRICT", Ref.keyword("CASCADE", optional=True), optional=True),
```
[the remark this answers is not in the document]
Comment (gimmyxd) [reply src/sqlfluff/dialects/dialect_ansi.py:2255]: @WittierDinosaur I wanted to be consistent with the other parts of this file, like: https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_ansi.py#L2395

I can remove `Ref.keyword` if it's cleaner without it.

[the remark this answers is not in the document]
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_ansi.py:2255]: Yeah - ideally, they wouldn't be there either

File: test/fixtures/dialects/ansi/drop_schema_b.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+drop schema if exists my_schema cascade
```
[the remark this answers is not in the document]
Comment (gimmyxd) [reply test/fixtures/dialects/ansi/drop_schema_b.sql]: Added some test cases

**Comment by WittierDinosaur** [comment]
Sorry for the nitpicking. But can you put all of the tests in the same file? You can have multiple tests in a file if they're delimited by ;

**Review by WittierDinosaur — APPROVED** [review]

**Pull request merged**